tools/system_libs.py

```diff
@@ -795,6 +793,11 @@ def get_non_lto_files(self):
     iprintf_files = files_in_path(
       path='system/lib/libc/musl/src/stdio',
       filenames=['__towrite.c', '__overflow.c', 'fwrite.c', 'fputs.c',
+                 'getc.c',
+                 'fputc.c',
+                 'fgets.c',
+                 'putc.c', 'putc_unlocked.c',
+                 'putchar.c', 'putchar_unlocked.c',
```

Comment:
Do I understand things right that files are put here when LLVM might emit them in LTO ops? That is, there is some LTO opt that could emit a call to `getc()` that did not exist before.

If so, might it be worth mentioning the source of such calls, like which LLVM pass does so? That might help us prune this list in the future. But if it's hard to do, it doesn't matter I suppose.

Reply:
The existing comment is probably enough for now. I guess if we need to find the specific LLVM passes that create these calls then we could search in the source.

Reply:
Maybe another option (instead of forcing these files to never be bitcode) is to compile both bitcode and native versions. The bitcode versions get linked into LTO as usual, but then we can add native versions to the native link, which would get pulled in if they weren't in the original link?

Reply:
That would require some kind of clever linker change though... as of today that linker always finds the first version of a given symbol and ignores all others.

Reply:
Another way of putting that would require the answer to the question "where is `putc` defined" to have a different answer before and after LTO.